### DISCUSSION TITLE
Where to add custom body classes to a page?

### DISCUSSION THREAD

**Discussion by banisterious** [GetPublii/Publii#1847] [Help] [answered]
Are custom classes for pages and posts supported out of the box, or would they need to be added to a config file or something? I'm guessing the latter. In any case, would appreciate suggestions. Goal is to add user-specified body classes to pages that would allow me to target specific pages in CSS.

**Comment by banisterious** [comment]
Whoops, I should have spent more time on my search. I think this is answered [here](https://github.com/GetPublii/Publii/discussions/1759#discussioncomment-10943647): "...add more classes by editing the theme files, especially the head.hbs"

**Reply by candidexmedia** [reply]
Also, always make sure to use theme overrides if you're working from an existing theme: https://getpublii.com/dev/theme-overrides/

**Comment by banisterious** [comment]
Since I'm here I may as well ask: How would I inject page slugs into custom classes in a handlebar file? That would probably be an ideal way to support my goal of being able to assign unique body classes.

**Reply by candidexmedia** [reply, marked as answer]
Use `{{id}}` or `{{slug}}` as the CSS ID

so, it could be something like...

```html
<element class="{{slug}}" id="{{id}}">
  ...
</element>
```

You can then target the element via the `#id` or `.class` in CSS

Doc ref: https://getpublii.com/dev/post-tags/


If you want the user to enter a custom name for each post's `<body>` class/ID name, you could do so via the [Post Settings API](https://getpublii.com/dev/theme-settings-api/#howtocreatepostsettings). Create a text field in the Post Settings, and call upon that setting in the handlebars template file (similar to how it's done for the ID / slug example above).
